clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -113,7 +120,7 @@ class EmitAssemblyHelper {
   const CodeGenOptions &CodeGenOpts;
   const clang::TargetOptions &TargetOpts;
   const LangOptions &LangOpts;
-  Module *TheModule;
+  llvm::Module *TheModule;
```

Comment:
Why did this suddenly need qualification?

Reply:
When including the new BackendConsumer.h header file to BackendUtils.cpp, we were indirectly including another header several layers deep that had defined clang::Module. Previously this file had no references to clang::Module via any includes, but with the new include we were getting an "ambiguous reference to Module" error. This was fixed by specifying which Module class we were using in this file.

I don't remember exactly which nested include referenced clang::Module, I could track down the include path again if it's helpful though.

Reply:
Without the qualification, we get the following:

In file included from llvm-project/llvm/include/llvm/IR/ModuleSummaryIndex.h:28,
                 from llvm-project/clang/include/clang/CodeGen/BackendUtil.h:13,
                 from llvm-project/clang/lib/CodeGen/BackendUtil.cpp:9:
llvm-project/llvm/include/llvm/IR/Module.h:65:32: note: candidates are: ‘class llvm::Module’
   65 | class LLVM_EXTERNAL_VISIBILITY Module {
      |                                ^~~~~~
In file included from llvm-project/clang/include/clang/Lex/ModuleLoader.h:18,
                 from llvm-project/clang/include/clang/Frontend/ASTUnit.h:26,
                 from llvm-project/clang/include/clang/Frontend/FrontendAction.h:23,
                 from llvm-project/clang/include/clang/CodeGen/CodeGenAction.h:12,
                 from llvm-project/clang/lib/CodeGen/BackendConsumer.h:13,
                 from llvm-project/clang/lib/CodeGen/BackendUtil.cpp:10:
llvm-project/clang/include/clang/Basic/Module.h:105:18: note:                 ‘class clang::Module’
  105 | class alignas(8) Module {
      |                  ^~~~~~